### PR TITLE
Add Streets Source v7 and v8 checks for way name layer

### LINKED ISSUE #1624: Upgrade to Mapbox Streets source v8
mapbox/mapbox-gl-native#13481 adds support for [Mapbox Streets source v8](https://www.mapbox.com/vector-tiles/mapbox-streets-v8/), which is currently in beta. By the time map SDK v7.0.0 comes out, we’ll need to add corresponding support for Streets source v8; otherwise, anyone using a v8-dependent style will incur twice as much overhead from vector tile downloading and processing, while map labels and the current road name label will go blank. We’ll want to retain support for v7 for the time being, since some popular navigation styles still use v7.

As far as I can tell, here are the changes we’ll need to make to accommodate both v7 and v8:

* [x] Add Streets source v8 instead of v7: https://github.com/mapbox/mapbox-navigation-android/blob/4ab858b4cad93b0abfa64b9074e38281787b97c8/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java#L69
* [x] Check if Streets source v8 is already part of the style, if so don't add: https://github.com/mapbox/mapbox-navigation-android/blob/4ab858b4cad93b0abfa64b9074e38281787b97c8/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java#L601-L602
* [x] If Streets source v7 is already part of the style, add v7 instead: https://github.com/mapbox/mapbox-navigation-android/blob/4ab858b4cad93b0abfa64b9074e38281787b97c8/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java#L601-L602
* [x] Use the `road` layer instead of `road_label` when the style contains Streets source v8: https://github.com/mapbox/mapbox-navigation-android/blob/4ab858b4cad93b0abfa64b9074e38281787b97c8/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java#L71

Unlike in mapbox/mapbox-navigation-ios#1896, it looks like the Android navigation SDK always adds a vector tile source for Streets source v8, without checking whether Streets source is already part of the style. That’s a problem because the SDK would end up downloading tiles for two different versions of the Streets source simultaneously, significantly increasing bandwidth consuption.

Does the Android navigation SDK automatically localize labels like on iOS? If so, there may need to be some changes to account for different language fallback mechanisms between v7 and v8.

/cc @mapbox/navigation-android @mapbox/maps-android @brsbl @taraniduncan

### PR DESCRIPTION
**Pull request by danesfeder**
Closes #1624 

Please see checklist in ticket for details.

cc @akitchen

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 1]
```diff
@@ -522,15 +526,33 @@ private void initializeMapLayerInteractor(MapboxMap mapboxMap) {
...
+      layerInteractor.addStreetsLayer(sourceV8.getId(), STREETS_V8_ROAD_LABEL);
+    } else {
+      VectorSource streetSource = new VectorSource(STREETS_SOURCE_ID, MAPBOX_STREETS_V8_URL);
+      mapboxMap.getStyle().addSource(streetSource);
+      layerInteractor.addStreetsLayer(STREETS_SOURCE_ID, STREETS_V8_ROAD_LABEL);
+    }
+  }
+
+  @Nullable
+  private Source findSourceById(List<Source> sources, String streetsUrl) {
```
Comment (1ec5) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: This method name threw me for a momentary loop: it’s actually finding the source by URL (specifically, by a tileset ID wrapped in a URL).

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 2]
```diff
@@ -522,15 +526,33 @@ private void initializeMapLayerInteractor(MapboxMap mapboxMap) {
...
+    for (Source source : sources) {
+      if (source instanceof VectorSource) {
+        VectorSource vectorSource = (VectorSource) source;
+        String url = vectorSource.getUrl();
+        if (url != null && url.contains(streetsUrl)) {
```
Comment (1ec5) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: A source URL might look like this and still be Streets source v8 for the purposes of this feature:

* mapbox://mapbox.mapbox-streets-v8
* mapbox://mapbox.mapbox-streets-v8,mapbox.mapbox-traffic-v1
* mapbox://mapbox.mapbox-streets-v8?foo=bar
* mapbox://mapbox.mapbox-streets-v8/?foo=bar

On iOS, we create a `URL` struct, check that the `scheme` is `mapbox`, and split the `host` on commas. Is it reasonably efficient to create a [`java.net.URL`](https://docs.oracle.com/javase/8/docs/api/java/net/URL.html) for each source URL?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java [hunk 3]
```diff
@@ -51,11 +49,11 @@
 public class NavigationMapboxMap {
 
   static final String STREETS_LAYER_ID = "streetsLayer";
-  private static final String MAPBOX_STREETS_V7 = "mapbox://mapbox.mapbox-streets-v7";
+  private static final String MAPBOX_STREETS_V7_URL = "mapbox://mapbox.mapbox-streets-v7";
+  private static final String MAPBOX_STREETS_V8_URL = "mapbox://mapbox.mapbox-streets-v8";
   private static final String STREETS_SOURCE_ID = "streetsSource";
```
Comment (1ec5) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java]: Might want to take this opportunity to make the source ID more unique, like `com.mapbox.navigation.streets`.

**Comment by danesfeder** [comment]
@1ec5 this is up-to-date and ready for another round when you have a cycle.

**Pull request merged**